mercure.rst

```diff
@@ -0,0 +1,564 @@
+    {
+        public function __invoke(Request $request): Response
+        {
+            $hubUrl = $this->getParameter('mercure.default_hub');
+            $this->addLink($request, new Link('mercure', $hubUrl));
+
+            $username = $this->getUser()->getUsername(); // Retrieve the username of the current user
+            $token = (new Builder())
+                // set other appropriate JWT claims, such as an expiration date
+                ->set('mercure', ['subscribe' => "http://example.com/user/$username"]) // could also include the security roles, or anything else
+                ->sign(new Sha256(), this->getParameter('mercure_secret_key')) // don't forget to set this parameter! Test value: aVerySecretKey
+                ->getToken();
+
+            $response = $this->json(['@id' => '/demo/books/1', 'availability' => 'https://schema.org/InStock']);
+            $response->headers->set(
+                'set-cookie',
+                sprintf('mercureAuthorization=%s; path=/hub; secure; httponly; SameSite=strict', $token)
+            );
+
+            return $response;
+        }
+    }
+
+.. caution::
+
+    To use the cookie authentication method, the Symfony app and the Hub
+    must be served from the same domain (can be different sub-domains).
+
+Generating Programmatically The JWT Used to Publish
+---------------------------------------------------
+
+Instead of directly storing a JWT in the configuration,
+you can create a service that will return the token used by
+the `Publisher` object.
+
+.. code-block:: php
+
+    // src/Mercure/MyJwtProvider.php
+    namespace App\Mercure;
+
+    final class MyJwtProvider
+    {
+        public function __invoke(): string
+        {
+            return 'the-JWT';
+        }
+    }
+
+Then, reference this service in the bundle configuration:
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        # config/packages/mercure.yaml
+        mercure:
+            hubs:
+                default:
+                    url: https://mercure-hub.example.com/hub
+                    jwt_provider: App\Mercure\MyJwtProvider
+
+    .. code-block:: xml
+
+        <!-- config/packages/mercure.xml -->
+        <?xml version="1.0" encoding="UTF-8" ?>
+        <config>
+            <hub
+                name="default"
+                url="https://mercure-hub.example.com/hub"
+                jwt-provider="App\Mercure\MyJwtProvider"
+            />
+        </config>
+
+    .. code-block:: php
+
+        // config/packages/mercure.php
+        use App\Mercure\MyJwtProvider;
+
+        $container->loadFromExtension('mercure', [
+            'hubs' => [
+                'default' => [
+                    'url' => 'https://mercure-hub.example.com/hub',
+                    'jwt_provider' => MyJwtProvider::class,
+                ],
+            ],
+        ]);
+
+This method is especially convenient when using tokens having an expiration
+date, that can be refreshed programmatically.
```

Comment:
> This method is especially convenient when using tokens having an expiration
> date, that can be refreshed programmatically.

When using tokens that have an expiration date, this method is convenient to refresh the tokens programmatically.

Reply:
It's not exactly the same meaning (some tokens having expiration dates cannot be refreshed, this is only useless for these ones)